### PR TITLE
Update categorization page

### PR DESCRIPTION
**Pull request by drazik**
Implementation of the new categorization pages.

### Desktop
screen shot 2018-05-16 at 14 25 53

### Mobile
screen shot 2018-05-16 at 14 27 16

### Desktop
screen shot 2018-05-16 at 14 28 19

### Mobile
screen shot 2018-05-16 at 14 27 52

### Desktop
screen shot 2018-05-16 at 14 28 57

### Mobile
screen shot 2018-05-16 at 14 29 25

### CODE REVIEW

File: src/ducks/categories/styles.styl [hunk 1]
```diff
@@ -19,25 +19,49 @@
             position absolute
             line-height 24px
 
-    .bnk-cat-figure
-        padding 2rem 0 0
-
     .bnk-cat-chart
-        flex-basis 50%
+        height 182px
```
Comment (ptbrowne) [inline src/ducks/categories/styles.styl]: just for information, why is this repeated here ?

File: src/ducks/categories/styles.styl [hunk 2]
```diff
@@ -19,25 +19,49 @@
...
         margin-top -100px
+        margin-left auto
```
Comment (ptbrowne) [inline src/ducks/categories/styles.styl]: why ?

File: src/ducks/categories/styles.styl [hunk 3]
```diff
@@ -19,25 +19,49 @@
...
+
+.CategoriesChart__FigureBlockContainer
+    position absolute
+    top 25px
+    bottom 25px
+    left 30px
+    right 30px
+    border-radius 50%
+    display flex
+    flexdrection column
```
Comment (ptbrowne) [inline src/ducks/categories/styles.styl]: typo :)

File: src/ducks/categories/styles.styl [hunk 4]
```diff
@@ -19,25 +19,49 @@
...
+    justify-content center
+    align-items center
+
+.CategoriesChart__FigureBlock
+    display flex
+    flex-direction column-reverse
```
Comment (ptbrowne) [inline src/ducks/categories/styles.styl]: joli :D

File: src/ducks/categories/styles.styl (hunk 2 above)
Comment (drazik) [reply src/ducks/categories/styles.styl]: To put it on the right of its flex container

File: src/ducks/categories/styles.styl (hunk 3 above)
Comment (drazik) [reply src/ducks/categories/styles.styl]: Thanks :+1:

File: src/ducks/categories/styles.styl (hunk 4 above)
Comment (drazik) [reply src/ducks/categories/styles.styl]: :sunglasses:

File: src/ducks/categories/CategoriesHeader.jsx [hunk 5]
```diff
@@ -43,7 +43,9 @@ const CategoriesHeader = ({
               checked={withIncome}
               onToggle={checked => onWithIncomeToggle(checked)}
             />
-            <label htmlFor="withIncome">Inclure les revenus</label>
+            <label htmlFor="withIncome">
+              {t('Categories.filter.includeIncome')}
+            </label>
```
Comment (ptbrowne) [inline src/ducks/categories/CategoriesHeader.jsx]: 😋

File: src/ducks/categories/Categories.jsx [hunk 6]
```diff
@@ -218,19 +243,30 @@ class Categories extends Component {
...
+                <Caption className={styles['bnk-table-row-caption']}>
+                  <span className={styles['bnk-table-percentage-caption']}>
+                    {!subcategory && selectedCategoryName
+                      ? '100%'
+                      : `${percentage}%`}
+                  </span>
+                  <span className={styles['bnk-table-operation-caption']}>
+                    {`${transactionsNumber} ${t(
+                      `Categories.headers.transactions.${
+                        transactionsNumber > 1 ? 'single' : 'plural'
```
Comment (ptbrowne) [inline src/ducks/categories/Categories.jsx]: FYI http://airbnb.io/polyglot.js/ has a smart_count feature. Not blocking at all.

File: src/ducks/categories/styles.styl [hunk 7]
```diff
@@ -32,6 +59,22 @@
...
+
+        &::after
+            content ''
+            display inline-block
+            height 1px
+            width 1px
+            background-color coolGrey
+            vertical-align middle
+            margin-left 5px
+            margin-right 4px
```
Comment (ptbrowne) [inline src/ducks/categories/styles.styl]: wha'ts this ?

File: src/components/Breadcrumb/index.jsx [hunk 8]
```diff
@@ -1,80 +1,93 @@
...
+        [styles['BreadcrumbItem--current']]: isCurrent
+      })}
     >
-      <Tag className={styles.Breadcrumb__title}>
-        {item.name}
-        {item.displaySeparator && (
-          <BreadcrumbSeparator separator={item.separator} />
-        )}
+      <Tag
+        onClick={onClick}
```
Comment (ptbrowne) [inline src/components/Breadcrumb/index.jsx]: Curious : why did you move the onClick handler ?

File: src/components/Breadcrumb/index.jsx [hunk 9]
```diff
@@ -1,80 +1,93 @@
...
+          tag={currentItem.tag}
+          isCurrent
+        />
+      </div>
     </div>
   )
 }
 
+Breadcrumb.propTypes = {
+  items: PropTypes.arrayOf(PropTypes.shape(itemPropTypes))
```
Comment (ptbrowne) [inline src/components/Breadcrumb/index.jsx]: nice.

File: src/components/Breadcrumb/style.styl [hunk 10]
```diff
@@ -1,31 +1,33 @@
...
+
+
+.Breadcrumb__previousButton
+    margin-right 1.5rem
+    cursor pointer
+
+.Breadcrumb__items
+    flex-grow 1
+    display flex
+    flex-direction column
```
Comment (ptbrowne) [inline src/components/Breadcrumb/style.styl]: Why the need for flex here ?

File: src/ducks/categories/CategoriesHeader.jsx [hunk 11]
```diff
@@ -24,13 +24,14 @@ const CategoriesHeader = ({
 }) => {
   const globalCurrency = getGlobalCurrency(categories)
   const transactionsTotal = getTransactionsTotal(categories)
+  const [previousItem] = breadcrumbItems.slice(-2, 1)
```
Comment (ptbrowne) [inline src/ducks/categories/CategoriesHeader.jsx]: Huh ?

```
[1,2,3].slice(-2, 1)[0]
undefined
[0, 1,2,3].slice(-2, 1)[0]
undefined
```

What is this supposed to do ?

Is it not simpler to do index access directly ?

```
const previousItem = breadcrumbItems[breadcrumbItems.length - 2] ?
```

File: src/ducks/categories/styles.styl (hunk 7 above)
Comment (drazik) [reply src/ducks/categories/styles.styl]: There's a veeeeery little dot between the percentage and the number of transactions : 

image

It's not dust on your screen, it's really a dot between elements. Don't try to remove it with your finger like I did :smile:

File: src/components/Breadcrumb/index.jsx (hunk 8 above)
Comment (drazik) [reply src/components/Breadcrumb/index.jsx]: Because if I put the `onClick` on the parent, it would also be on the separator. Which seems weird to me.

File: src/components/Breadcrumb/style.styl (hunk 10 above)
Comment (drazik) [reply src/components/Breadcrumb/style.styl]: That's totally useless actually. I will remove it. Thanks

**Comment by drazik** [comment]
Waiting for Flo & Joël design review before merging, so I can do some little fixes in the context of this PR.

File: src/ducks/categories/styles.styl (hunk 7 above)
Comment (ptbrowne) [reply src/ducks/categories/styles.styl]: lol. I think https://fr.wikipedia.org/wiki/Point_m%C3%A9dian is what needs to be used ;)

File: src/ducks/categories/helpers.js [hunk 12]
```diff
@@ -125,3 +125,48 @@ export const isHealth = transaction => {
...
+      category.subcategories = category.subcategories.sort(subCategorySort)
+      transactionsTotal += category.amount
+    }
+  }
+
+  return transactionsTotal
+}
+
+export const getGlobalCurrency = categories =>
+  categories.length > 0 ? categories[0].currency : '€'
```
[the remark this answers is not in the document]
Comment (ptbrowne) [reply src/ducks/categories/helpers.js]: C'est pas categories[0].operations[0].currency ?
